host/sgx/sgx_enclave_common.c

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include "sgx_enclave_common.h"
+#include <openenclave/internal/trace.h>
+#include <stdlib.h>
+#include "../dynload.h"
+#include "../hostthread.h"
+
+/****** Pointers to functions in libsgx_enclave_common.so **************/
+static void* COMM_API (*_enclave_create)(
+    COMM_IN_OPT void* base_address,
+    COMM_IN size_t virtual_size,
+    COMM_IN size_t initial_commit,
+    COMM_IN uint32_t type,
+    COMM_IN const void* info,
+    COMM_IN size_t info_size,
+    COMM_OUT_OPT uint32_t* enclave_error);
+
+static size_t COMM_API (*_enclave_load_data)(
+    COMM_IN void* target_address,
+    COMM_IN size_t target_size,
+    COMM_IN_OPT const void* source_buffer,
+    COMM_IN uint32_t data_properties,
+    COMM_OUT_OPT uint32_t* enclave_error);
+
+bool COMM_API (*_enclave_initialize)(
+    COMM_IN void* base_address,
+    COMM_IN const void* info,
+    COMM_IN size_t info_size,
+    COMM_OUT_OPT uint32_t* enclave_error);
+
+bool COMM_API (*_enclave_delete)(
+    COMM_IN void* base_address,
+    COMM_OUT_OPT uint32_t* enclave_error);
+
+static bool COMM_API (*_enclave_set_information)(
+    COMM_IN void* base_address,
+    COMM_IN uint32_t info_type,
+    COMM_IN void* input_info,
+    COMM_IN size_t input_info_size,
+    COMM_OUT_OPT uint32_t* enclave_error);
+
+/****** Dynamic loading of libsgx_enclave_common.so/.dll **************/
+static void* _module;
+
+static void _unload_libsgx_enclave_common(void)
+{
+    if (_module)
+        oe_shared_library_unload(_module);
```

Comment:
*Non-blocking* Although based on the flow, there should not be the case that calling `_load_libsgx_enclave_common` after this call. But may be explicitly set _module = `NULL` here could help with eliminating potential warning.

Reply:
Addresses in PR #3155